src/ContainerInterface.php

```diff
@@ -12,12 +12,14 @@ interface ContainerInterface
     /**
      * Finds an entry of the container by its identifier and returns it.
      *
-     * @param string $id Identifier of the entry to look for.
+     * @template T of object
+     *
+     * @param class-string<T>|string $id Identifier of the entry to look for.
      *
      * @throws NotFoundExceptionInterface  No entry was found for **this** identifier.
      * @throws ContainerExceptionInterface Error while retrieving the entry.
      *
-     * @return mixed Entry.
+     * @return T|mixed Entry.
```

Comment:
Maybe would be better the response:
`@return ($id is class-string<T> ? T : mixed)`

PHPStan documentation: https://phpstan.org/blog/generics-by-examples#function-accepts-any-string%2C-but-returns-object-of-the-same-type-if-it%E2%80%99s-a-class-string
PSALM documentation: https://psalm.dev/docs/annotating_code/type_syntax/conditional_types/

Reply:
I have use same return type and works amazing with @phpstan as [stub](https://phpstan.org/user-guide/stub-files)

Reply:
done :)